src/h5web/vis-packs/core/gammaScale.ts

```diff
@@ -0,0 +1,151 @@
+import type {
+  InterpolatorFactory,
+  NumberValue,
+  ScalePower,
+  UnknownReturnType,
+} from 'd3-scale';
+import { isNumber } from 'lodash';
+
+function normalize(a: number, b: number) {
+  const extent = b - a;
+  return (x: number) => (extent !== 0 ? (x - a) / extent : 0.5);
+}
+
+type Range = number;
+type Output = Range;
+type Unknown = never;
+export type GammaScale = ScalePower<Range, Output, Unknown>;
+
+export function gammaScale(config?: GammaScaleConfig): GammaScale {
+  let _domain: number[] = config?.domain || [0, 1];
+  let _range: Range[] = config?.range || [0, 1];
+  let _exponent = config?.exponent ?? 1;
+  let _clamp = config?.clamp || false;
+  let _interpolate: InterpolatorFactory<Range, Output> = interpolateNumber;
+  let _unknown: Unknown;
+
+  const scaleFn = function (val: NumberValue) {
+    const [d0, d1] = _domain;
+    const [r0, r1] = _range;
+    const x = isNumber(val) ? val : val.valueOf();
+
+    return _interpolate(r0, r1)(normalize(d0, d1)(clamper(x)) ** _exponent);
+  };
+
+  function clamper(x: number) {
+    if (!_clamp) {
+      return x;
+    }
+    const [d0, d1] = _domain;
+    return Math.max(d0, Math.min(d1, x));
+  }
+
+  function domain(): Domain;
+  function domain(d: Domain): typeof scale;
+  function domain(d?: Domain) {
+    if (d) {
+      _domain = d;
+      return scale;
+    }
+    return [..._domain];
+  }
+
+  function range(): Range[];
+  function range(r: Range[]): typeof scale;
+  function range(r?: Range[]) {
+    if (r) {
+      _range = r;
+      return scale;
+    }
+    return [..._range];
+  }
+
+  function rangeRound(): Range[];
+  function rangeRound(r: Range[]): typeof scale;
+  function rangeRound(r?: Range[]) {
+    if (r) {
+      _range = r;
+      _interpolate = interpolateRound;
+      return scale;
+    }
+    return [..._range];
+  }
+
+  function clamp(): boolean;
+  function clamp(c: boolean): typeof scale;
+  function clamp(c?: boolean) {
+    if (c) {
+      _clamp = c;
+      return scale;
+    }
+    return _clamp;
+  }
+
+  function unknown(): UnknownReturnType<Unknown, never>;
+  // function unknown(u: Unknown): typeof scale;
+  function unknown() {
+    // if (u) {
+    //   _unknown = u;
+    //   return scale;
+    // }
+    return _unknown;
+  }
+
+  function interpolate(): InterpolatorFactory<Range, Output>;
+  function interpolate(i: InterpolatorFactory<Range, Output>): typeof scale;
+  function interpolate(i?: InterpolatorFactory<Range, Output>) {
+    if (i) {
+      _interpolate = i;
+      return scale;
+    }
+    return _interpolate;
+  }
+
+  function exponent(): number;
+  function exponent(e: number): typeof scale;
+  function exponent(e?: number) {
+    if (e) {
+      _exponent = e;
+      return scale;
+    }
+    return _exponent;
+  }
+
+  function invert(val: Output) {
+    const [d0, d1] = _domain;
+    const [r0, r1] = _range;
+
+    return clamper(
+      interpolateNumber(d0, d1)(normalize(r0, r1)(val) ** (1 / _exponent))
```

Comment:
Note that I use `interpolateNumber` here and not `_interpolate`. The rounding is indeed done when scaling from `domain` to `range` but not the other way round.